exec.go

```diff
@@ -46,23 +44,16 @@ func ExecCommand(ui Ui, input ExecCommandInput) {
 		ui.Error.Fatal(err)
 	}
 
-	cfg, err := writeTempConfig(input.Profile, profs)
-	if err != nil {
-		ui.Error.Fatal(err)
-	}
-
-	env := os.Environ()
-	env = overwriteEnv(env, "AWS_CONFIG_FILE", cfg.Name())
-	env = overwriteEnv(env, "AWS_DEFAULT_PROFILE", input.Profile)
-	env = overwriteEnv(env, "AWS_PROFILE", input.Profile)
-
-	env = unsetEnv(env, "AWS_ACCESS_KEY_ID")
-	env = unsetEnv(env, "AWS_SECRET_ACCESS_KEY")
-	env = unsetEnv(env, "AWS_CREDENTIAL_FILE")
+	env := environ(os.Environ())
+	env.Set("AWS_CONFIG_FILE", "/dev/null")
+	env.Set("AWS_VAULT", input.Profile)
```

Comment:
This paired with the removal of the generated config file will make for a 3.x.x release.